packages/nestjs-rate-limit/src/rate-limit.guard.ts

```diff
@@ -0,0 +1,45 @@
+import { CanActivate, ExecutionContext, HttpException, HttpStatus, Inject, Injectable } from '@nestjs/common';
+import assert from 'assert/strict';
+import { Request, Response } from 'express';
+
+import { Providers } from './constants';
+import { RateLimitOptions } from './interfaces/rate-limit-options.interface';
+import { TokenExtractor } from './interfaces/token-extractor.interface';
+import { RateLimitService } from './rate-limit.service';
+
+@Injectable()
+export class RateLimitGuard implements CanActivate {
+  private static extractTokenFromHeadersOrCookies(executionContext: ExecutionContext): string | undefined {
+    const request = executionContext.switchToHttp().getRequest<Request>();
+
+    // TODO: Pretty sure this is not how we're meant to access the cookies
+    return request.headers.authorization || request.cookies.auth_vf;
```

Comment:
please lmk if this is okay

Reply:
Seems fine to me, the cookie seems to just be the token as well.

[Looks like nest](https://docs.nestjs.com/techniques/cookies#use-with-express-default) still requires the `cookie-parser` package for the `cookies` prop to be available though.